pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -322,14 +403,33 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 			return nil
 		} else /* pv != nil */ {
 			// Found a PV for this claim
+			// There are three scenarios here:
+			// 1. the PV is newly dynamically provisioned
+			// 2. an existing PV meets the requirement of the PVC and available
+			// 3. user set "claimRef" of a PV to the PVC
+			// for metric reporting, the first case will be provision operation latency, and the
+			// other two would be binding, if there exists NO provision timestamp AND the volume
+			// got NO annotation of pvutil.annDynamically, its a binding operation, insert
+			// a timestamp for start bindings
+			claimKey := claimToClaimKey(claim)
+			metricOperation := "provision"
+			if !ctrl.existsOperationTimestamp("provision", claimKey) && !metav1.HasAnnotation(volume.ObjectMeta, pvutil.AnnDynamicallyProvisioned) {
+				ctrl.addOperationTimestamp("binding", claimKey, ctrl.getProvisionerNameFromVolume(volume))
+				metricOperation = "binding"
```

Comment:
So if I understand this correctly, if volume was not pre-provisioned(there was a free PV around) then there won't be a "binding" metric for that volume? 

IMO - we should just report provision+binding metric as one data point for volumes which are dynamically provisioned. I doubt people care about binding and pre-binding metrics and having that code here makes is more complex.

Reply:
I agree the code will be slightly more compared to just recording the provision latency, however with reasonable considerations. Standing from a user's perspective, prebinding/binding metric do have good potential uses. I'd imagine applications who care about bootstrapping time will certainly try to have volume pre-provisioned for use, and uses prebinding/binding to establish SLOs around them. Also, statically provisioned volumes are widely used, users would still like to have an idea about how much time was spent for k8s to bind/attach/mount the volume to their pvc/pod. Thus the implementation includes those different operations.

With the combination of binding/pre-binding, the code will look a bit simpler as well.

Reply:
no longer apply.
per discussion with @gnufied @msau42 
prebinding/binding latency/error count metric recording can be a future enhancement but will not be introduced within this change